src/IoTeX/Staking.cpp

```diff
@@ -5,56 +5,120 @@
 // file LICENSE at the root of the source code distribution tree.
 
 #include "Staking.h"
-
-#include "Ethereum/ABI/Function.h"
-#include "Data.h"
-#include "uint256.h"
+#include "HexCoding.h"
+#include <TrustWalletCore/TWData.h>
+#include <TrustWalletCore/TWString.h>
+using namespace TW;
 
 namespace TW::IoTeX {
 
-using namespace TW::Ethereum::ABI;
-
-void stakingStake(const Data& candidate, uint64_t stakeDuration, bool nonDecay, const Data& dataIn, Data& dataOut) {
-    Function func("createPygg");
-    func.addInParam(std::make_shared<ParamByteArrayFix>(12, candidate));
-    func.addInParam(std::make_shared<ParamUInt256>(uint256_t(stakeDuration)));
-    func.addInParam(std::make_shared<ParamBool>(nonDecay));
-    func.addInParam(std::make_shared<ParamByteArray>(dataIn));
+const char* charFromTWData(TWData* _Nonnull data) {
+    return TWStringUTF8Bytes(TWStringCreateWithRawBytes(TWDataBytes(data), TWDataSize(data)));
+}
 
-    func.encode(dataOut);
+TWData* _Nullable stakingCreate(TWData* _Nonnull candidate, TWData* _Nonnull amount,
```

Comment:
`TWData` is C struct, you should use `Data` here

Reply:
sure